pontoon/sync/changeset.py

```diff
@@ -108,7 +109,40 @@ def execute(self):
         self.bulk_create_translations()
         self.bulk_update_translations()
 
+        # Clean up any duplicate approvals
+        with connection.cursor() as cursor:
+            cursor.execute("""
+                UPDATE base_translation AS b
+                SET approved = FALSE, approved_date = NULL
+                WHERE
+                    id IN
+                    (SELECT trans.id FROM base_translation AS trans
+                        LEFT JOIN base_entity AS ent ON ent.id = trans.entity_id
+                        LEFT JOIN base_resource AS res ON res.id = ent.resource_id
+                        WHERE locale_id = %(locale_id)s
+                        AND res.project_id = %(project_id)s)
+                    AND approved_date !=
+                    (SELECT max(approved_date)
+                        FROM base_translation
+                        WHERE entity_id = b.entity_id
+                        AND locale_id = b.locale_id
+                        AND (plural_form = b.plural_form OR plural_form IS NULL));
+            """, {
+                'locale_id': self.locale.id,
```

Comment:
I'm getting a NoneType error when syncing with Git repo from `self.locale.id`. DB migrations have been done. Am I missing something when trying to upgrade my dev instance?

Reply:
Should be fixed in https://github.com/mozilla/pontoon/commit/20bffedb82441e4c7fe5d2813691627221575054. Let me know if the problem persist.

Reply:
Thanks for the quick reply @mathjazz , I guess I should have mentioned from the start that the issue appeared for me in 20bffedb82441e4c7fe5d2813691627221575054 . I then dropped to 6db15d17c9707ac1340119acc0e67dba7839fac0 to try if it works in the original merge - it didn't. 😭

Reply:
Hm, that's surprising. Could you paste the stack trace?

Reply:
I tried reproducing this issue for the stack trace going from safe state e2fbf0cbeb09cb12e8ea373768dffb9a2101b3e0 to 20bffedb82441e4c7fe5d2813691627221575054 and no longer couldn't reproduce it. Issue solved. Thanks!